### ISSUE TITLE
Change pyright CI to utilise poetry instance

### ISSUE THREAD

**Issue by SimonBiggs** [pymedphys/pymedphys#1703]
That way pyright will obey the `poetry.lock` file and not cause CI failures until we explicitly undergo an update cycle.

**Comment by SimonBiggs** [comment]
If someone was keen to give this a shot, that'd be awesome :)

**Comment by SimonBiggs** [comment]
@ddcc3388,

Let me know if you were interested in learning how this is done. I could jump on a video call with you and show you how to do this.

**Comment by ddcc3388** [comment]
@SimonBiggs Hey Simon, I'm still trying to sort out the contributor's section. Would you have time to show me tomorrow?

**Comment by SimonBiggs** [comment]
> Would you have time to show me tomorrow?

Of course. Let's book either early morning or late afternoon so that I can have a good block on uninterrupted coding time myself.

**Comment by ddcc3388** [comment]
Great! We could do late afternoon if that works. What time suits you?

**Comment by SimonBiggs** [comment]
Might we be able to do 5pm? (AEST)
